pkg/validate/validate.go

```diff
@@ -115,13 +115,23 @@ func ClusterName1035(v string) error {
 	return ClusterName(v)
 }
 
+// ClusterName1035WithMaxLength validates if the string provided length is
+// greater than maxlen argument.
+func ClusterName1035WithMaxLength(v string, maxlen int) error {
```

Comment:
This function is not doing RFC-1035 validation. Change the name to reflect what the function is actually doing.